File: server/router/system/phone_register_task.go
package system

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type PhoneRegisterTaskRouter struct{}

func (r *PhoneRegisterTaskRouter) InitPhoneRegisterTaskRouter(PrivateGroup *gin.RouterGroup, PublicGroup *gin.RouterGroup) {
	phoneRegisterTaskRouter := PrivateGroup.Group("phoneRegisterTask").Use(middleware.OperationRecord())
	phoneRegisterTaskRouterWithoutRecord := PrivateGroup.Group("phoneRegisterTask")
	publicPhoneRegisterTaskRouter := PublicGroup.Group("phoneRegisterTask")
	{
		phoneRegisterTaskRouter.POST("create", phoneRegisterTaskApi.CreatePhoneRegisterTask)
		phoneRegisterTaskRouter.POST("submitCode", phoneRegisterTaskApi.SubmitPhoneRegisterTaskCode)
		phoneRegisterTaskRouter.POST("settle", phoneRegisterTaskApi.SettlePhoneRegisterTaskLeader)
	}
	{
		phoneRegisterTaskRouterWithoutRecord.GET("active", phoneRegisterTaskApi.GetActivePhoneRegisterTask)
		phoneRegisterTaskRouterWithoutRecord.GET("actives", phoneRegisterTaskApi.GetActivePhoneRegisterTasks)
		phoneRegisterTaskRouterWithoutRecord.GET("submitStatus", phoneRegisterTaskApi.GetPhoneRegisterSubmitStatus)
		phoneRegisterTaskRouterWithoutRecord.POST("list", phoneRegisterTaskApi.GetPhoneRegisterTaskList)
		phoneRegisterTaskRouterWithoutRecord.POST("logs", phoneRegisterTaskApi.GetPhoneRegisterTaskLogs)
		phoneRegisterTaskRouterWithoutRecord.GET("summary", phoneRegisterTaskApi.GetPhoneRegisterTaskSummary)
		phoneRegisterTaskRouterWithoutRecord.GET("settlement/history", phoneRegisterTaskApi.GetPhoneRegisterTaskSettlementHistory)
	}
	{
		publicPhoneRegisterTaskRouter.POST("device/poll", phoneRegisterTaskApi.DevicePollPhoneRegisterTask)
		publicPhoneRegisterTaskRouter.GET("device/task", phoneRegisterTaskApi.DeviceGetPhoneRegisterTask)
		publicPhoneRegisterTaskRouter.POST("device/task", phoneRegisterTaskApi.DeviceGetPhoneRegisterTask)
		publicPhoneRegisterTaskRouter.POST("device/heartbeat", phoneRegisterTaskApi.DeviceHeartbeatPhoneRegisterTask)
		publicPhoneRegisterTaskRouter.POST("device/report", phoneRegisterTaskApi.DeviceReportPhoneRegisterTask)
		publicPhoneRegisterTaskRouter.POST("device/log", phoneRegisterTaskApi.DeviceLogPhoneRegisterTask)
		publicPhoneRegisterTaskRouter.GET("device/config", phoneRegisterTaskApi.DeviceGetPhoneRegisterConfig)
		publicPhoneRegisterTaskRouter.POST("open-api/task", phoneRegisterTaskApi.OpenAPIPollPhoneRegisterTask)
		publicPhoneRegisterTaskRouter.GET("open-api/task", phoneRegisterTaskApi.OpenAPIGetPhoneRegisterTask)
		publicPhoneRegisterTaskRouter.POST("open-api/verify-code", phoneRegisterTaskApi.OpenAPIGetVerifyCode)
		publicPhoneRegisterTaskRouter.POST("open-api/report", phoneRegisterTaskApi.OpenAPIReportPhoneRegisterTask)
		publicPhoneRegisterTaskRouter.POST("open-api/cache", phoneRegisterTaskApi.OpenAPIUploadPhoneRegisterCache)
		publicPhoneRegisterTaskRouter.GET("open-api/promoter/device-stats", phoneRegisterTaskApi.PromoterOpenAPIDeviceStats)
		publicPhoneRegisterTaskRouter.POST("open-api/promoter/task", phoneRegisterTaskApi.PromoterOpenAPICreateTask)
	}
}

File: web/src/api/phoneRegisterTask.js
import service from '@/utils/request'

export const createPhoneRegisterTask = (data) => {
  return service({
    url: '/phoneRegisterTask/create',
    method: 'post',
    data
  })
}

export const submitPhoneRegisterTaskCode = (data) => {
  return service({
    url: '/phoneRegisterTask/submitCode',
    method: 'post',
    data
  })
}

export const getActivePhoneRegisterTask = (config = {}) => {
  return service({
    url: '/phoneRegisterTask/active',
    method: 'get',
    ...config
  })
}

export const getActivePhoneRegisterTasks = (config = {}) => {
  return service({
    url: '/phoneRegisterTask/actives',
    method: 'get',
    ...config
  })
}

export const getPhoneRegisterSubmitStatus = (config = {}) => {
  return service({
    url: '/phoneRegisterTask/submitStatus',
    method: 'get',
    ...config
  })
}

export const getPhoneRegisterTaskList = (data, config = {}) => {
  return service({
    url: '/phoneRegisterTask/list',
    method: 'post',
    data,
    ...config
  })
}

export const getPhoneRegisterTaskSummary = (params) => {
  return service({
    url: '/phoneRegisterTask/summary',
    method: 'get',
    params
  })
}

export const settlePhoneRegisterTaskLeader = (data) => {
  return service({
    url: '/phoneRegisterTask/settle',
    method: 'post',
    data
  })
}

export const getPhoneRegisterTaskSettlementHistory = (params) => {
  return service({
    url: '/phoneRegisterTask/settlement/history',
    method: 'get',
    params
  })
}

export const getPhoneRegisterTaskLogs = (data) => {
  return service({
    url: '/phoneRegisterTask/logs',
    method: 'post',
    data
  })
}
